### ISSUE TITLE
jwt import unrecognized despite usage and cannot use sign() due to TypeError

### ISSUE THREAD

**Issue by mslipenchuk267** [zaguiini/react-native-pure-jwt#23]
**Describe the bug**
I am simply trying to encode a JWT using the sign function, but when I import jwt from 'react-native-pure-jwt' my editor tells me that jwt is declared but never used, even though I am clearly using it as instructed in the usage section. When the code arrives at the line where I sign a token I get the following error:
```
TypeError: _reactNativePureJwt.default.sign is not a function. (In '_reactNativePureJwt.default.sign(payload, "s3cr3tt", {
              alg: "HS256"
            })', '_reactNativePureJwt.default.sign' is undefined)
```

**To Reproduce**
1. Import jwt from "react-native-pure-jwt" at the top of a react native functional component file
2. Make a button that calls the sign() function when it is pressed
3. Observe the error when the function is called due to sign()

**Expected behavior**
the sign() function should return a string corresponding to the JWT.

**Screenshots**
Code showing that jwt is unused even though I am using it:
![Screen Shot 2020-10-30 at 11 59 34 AM](https://user-images.githubusercontent.com/25811429/97727988-6289f100-1aa7-11eb-8cd2-82f815580c93.png)
The error given in console:
![Screen Shot 2020-10-30 at 11 59 57 AM](https://user-images.githubusercontent.com/25811429/97728028-6fa6e000-1aa7-11eb-830a-3dd27f3b2460.png)

**Desktop**
 - OS: macOS v10.15.7
 - react-native: 0.63.2
 - react-native-cli: 2.0.1

**Smartphone (please complete the following information):**
 - Device: iPhone8
 - OS: iOS14

**Additional context**
The application builds fine and I get this same error when trying auto or manual linking as described in the install instructions.

**Comment by osancus** [comment]
having the same issue as @mslipenchuk267

**Comment by zaguiini** [comment]
Hm, that's very weird. Have you guys managed to fix the problem? If yes, what was it?

**Comment by mslipenchuk267** [comment]
No I still have the issue, could your reproduce my problem?

**Comment by zaguiini** [comment]
Unfortunately, I couldn't. 😞 

It works on my machine, even though I could reproduce the unused import false flag!

My first impression is that it's a TypeScript check issue on VSCode specifically, as it doesn't yield the same error on WebStorm/IDEA:

![image](https://user-images.githubusercontent.com/26530524/99526117-95722700-2979-11eb-81ce-85d0b3cd0448.png)

See, it's not an ESLint complaint, but a `ts` one. I wonder if this is reported somewhere else... This is probably not an issue with `react-native-pure-jwt`.

---

This is how I created a token:

![image](https://user-images.githubusercontent.com/26530524/99526562-2f39d400-297a-11eb-8b4c-3917777ceda5.png)

And it works properly:

![image](https://user-images.githubusercontent.com/26530524/99526593-37920f00-297a-11eb-80f6-2df1d9d87073.png)

Using:

```
    "react-native": "0.63.3",
    "react-native-pure-jwt": "^2.2.2"
```

on an iPhone 8, running iOS 14.2.

**Comment by zaguiini** [comment]
One question: are you sure you're not messing with imports in some way? Post-processing on babel or stuff? Or are you using TypeScript?

From what I see it reports that it wraps the import within a `default` object (`jwt.default.sign`) instead of consuming as `jwt.sign`.

Have you tried importing it like `import * as jwt from 'react-native-pure-jwt'`?

**Comment by mslipenchuk267** [comment]
Hmm I don't think so, I don't use Typescript. 

I actually had to change my solution to meet a deadline but I will try my best to test this out this following week and update.

**Comment by Salman-Shaik** [comment]
Any Solution, I'm still facing this problem.

**Comment by billnbell** [comment]
I have a solution for this

**Comment by sharesoftinfo** [comment]
after your changes did.... still sign() method not work...my app crashed. kindly help me!

**Comment by aliraza-noon** [comment]
hello all, I also experienced this problem make sure you have installed the pods

**Comment by akshaypunchh** [comment]
Hello @mslipenchuk267 , You have to miss to install pod.

**Comment by vitara-slipenchuk** [comment]
> Hello @mslipenchuk267 , You have missed to install pod. 

Hmm I don't think thats the problem as I tried pod install with automatic and manual linking but at the time it didn't work either way. I mentioned this in the Additional Context section of my post.

**Comment by mslipenchuk267** [comment]
> Hello @mslipenchuk267 , You have missed to install pod. 

I don't believe that's the issue as I tried both auto and manual linking of the library when trying to install. I mentioned this in the Additional Context Section of my post.

**Comment by akshaypunchh** [comment]
Hello @mslipenchuk267 , 

If you still facing same error, you can connect me on skype (akshay.punchh). I've used a workaround to fix it.

**Comment by juanlatorre** [comment]
Hey, I'm still having this issue, I can't find any workaround

**Comment by stryaponoff** [comment]
Fixed version is already in the master branch but not in npm. I've solved the problem by replacing this file https://github.com/zaguiini/react-native-pure-jwt/blob/master/js/index.d.ts using patch-package.

@zaguiini could you push the latest version to npm?

**Comment by JulioOrellana** [comment]
Hello guys, the issue I found is because a JWT dependency in this lib (at least on iOS). Since dependency on podfile is set to `~> 3.0.0-beta.11`,  the version actually installed is `3.0.0-beta.14` so it returns this error:

```swift
.../node_modules/react-native-pure-jwt/ios/RNPureJwt.m:53:43: error: property 'claimsSet' not found on object of type 'JWTEncodingBuilder *'
    JWTCodingResultType *result = builder.claimsSet(claimsSet).addHolder(holder).result;
                                          ^
.../node_modules/react-native-pure-jwt/ios/RNPureJwt.m:81:80: error: property 'claimsSet' not found on object of type 'JWTDecodingBuilder *'
    JWTCodingBuilder *verifyBuilder = [JWTDecodingBuilder decodeMessage:token].claimsSet(trustedClaims).addHolder(holder);
                                                                               ^
.../node_modules/react-native-pure-jwt/ios/RNPureJwt.m:88:54: error: property 'claimsSet' not found on object of type 'JWTCodingResultTypeSuccess *'
        NSDictionary *claims = [result.successResult.claimsSet dictionaryWithValuesForKeys: claimsKeys];
```

To fix this issue on iOS only, I have not tested on Android yet, [patch-package](https://www.npmjs.com/package/patch-package) -> `react-native-pure-jwt.podspec` and change the dependency to `s.dependency 'JWT', '3.0.0-beta.12'`, run `pod install --repo-update` or `pod update JWT` and reinstall the app

Until `3.0.0-beta.12` (JWTDecodingBuilder|JWTEncodingBuilder|JWTCodingResultTypeSuccess) supports `claimsSet`, higher than that returns the error

**Comment by bolddp** [comment]
Until the fix is published to npm, you can install it with
```
> npm install zaguiini/react-native-pure-jwt
```

Not sure how to do this with yarn...

**Comment by zaguiini** [comment]
Sorry for not having this updated for so long. It's published now under version `2.2.3`. Please update.
